bool initialized_
uint16 data